NuRadioMC/simulation/simulation.py

```diff
@@ -813,6 +830,15 @@ def run(self):
                         tmp = tmp[:-2] + " ns"
                         logger.info(f"creating event {iEvent} of event group {self._event_group_id} ranging rom {iStart} to {iStop} with indices {indices} corresponding to signal times of {tmp}")
                     self._evt = NuRadioReco.framework.event.Event(self._event_group_id, iEvent)  # create new event
+
+                    # add MC particles that belong to this (sub) event to event structure
+                    # add only primary and parent?
+                    # or all particles? usually not so many in our case...
+                    self._evt.add_particle(self.primary)
```

Comment:
yes we want the primary particle, and all parents I think. but for everything we do right now, there is only one or at most two parents. For muon or tau emission it would be "neutrino" -> "tau" -> "shower". For the latter case, how would we get this information? 
Also, don't you need to also add the current particle (which is then actually a shower)?

Reply:
as far as i can see this point maybe needs more discussion...
* Do we want to store all showers as a particle also, in principle, or does this just duplicate info in the sim showers?
  Note: I already added a parent_id to the showers, so that one can look at the sim showers and look for their parents, via the 
   event.get_parents(particle_or_shower) or via get_interaction_products(self, parent_particle, showers = True, particles = True) return the daughter particles/showers/both. Maybe you have a suggestion for a more clever way?
* In principle one should store all the particles, but I would maybe argue that one should only ever store particles passed by the generator? I.e. I would not generate outgoing leptons with (1-y) within simulation.py

* I propose: For every event group id, read neutrino properties from entry 0, then In the input shower/particle loop in simulation.py check, if the line corresponds to a shower or particle
* if particle, add particle to the list and continue to next item; if shower use the current looping over stations in simulation.py

Reply:
One more thing I add here so that I remember: maybe in the particle instead of `flavor` it would be better to store the PDG code...Makes more sense for e/mu/tau leptons, and possibly a custom code for showers could be defined.